infra/modules/cicd/main.tf

```diff
@@ -80,43 +116,134 @@ resource "google_cloudbuild_trigger" "app_new_build" {
   pubsub_config {
     topic = google_pubsub_topic.gcr.id
   }
+
   approval_config {
-    approval_required = true
-  }
-  substitutions = {
-    _SERVICE             = var.run_service_name
-    _IMAGE_NAME          = "$(body.message.data.tag)"
-    _REGION              = var.region
-    _RUN_SERVICE_ACCOUNT = local.run_service_account
+    approval_required = false
   }
+
   source_to_build {
     uri       = local.github_repository_url
     ref       = "refs/heads/main"
     repo_type = "GITHUB"
   }
-  git_file_source {
-    path      = "build/app-deploy.cloudbuild.yaml"
-    repo_type = "GITHUB"
-    revision  = "refs/heads/main"
-    uri       = local.github_repository_url
+
+  build {
+    images = []
+    substitutions = {
+      _REGION           = var.region
+      _RUN_SERVICE_NAME = var.run_service_name
+      _PIPELINE_NAME    = google_clouddeploy_delivery_pipeline.default.name
+      _IMAGE            = "${local.artifact_registry_repo}/app"
+      _RUN_CONFIG       = local.run_config
+      _SKAFFOLD_CONFIG  = local.skaffold_config
+    }
+    tags = []
+    options {
+      logging = "CLOUD_LOGGING_ONLY"
+    }
+    dynamic "step" {
+      for_each = yamldecode(local.new_release_config).steps
+      content {
+        args       = step.value.args
+        name       = step.value.name
+        entrypoint = step.value.entrypoint
+        id         = step.value.id
+      }
+    }
+  }
+}
+# Cloud Deploy resources (pipeline, target service account, and IAM bindings)
+resource "google_clouddeploy_delivery_pipeline" "default" {
+  project     = var.project_id
+  location    = var.region
+  name        = "${var.deployment_name}-delivery"
+  description = "Basic delivery pipeline for ${var.deployment_name} app."
+  labels      = var.labels
+  serial_pipeline {
+    stages {
+      profiles  = ["prod"]
+      target_id = google_clouddeploy_target.prod.name
+    }
   }
+  depends_on = [
+    time_sleep.project_services
+  ]
+}
+
+resource "google_service_account" "cloud_deploy" {
+  project      = var.project_id
+  account_id   = "${var.deployment_name}-cloud-deploy"
+  display_name = "Service Account for Cloud Deploy deployment to Cloud Run."
 }
 
+resource "google_project_iam_member" "deploy_job_runner" {
+  project = var.project_id
+  role    = "roles/clouddeploy.jobRunner"
+  member  = "serviceAccount:${google_service_account.cloud_deploy.email}"
+}
+
+resource "google_project_iam_member" "deploy_run_admin" {
+  project = var.project_id
+  role    = "roles/run.admin"
+  member  = "serviceAccount:${google_service_account.cloud_deploy.email}"
+}
+
+resource "google_service_account_iam_binding" "deploy_sa_user_run" {
+  service_account_id = data.google_service_account.cloud_run.id
+  role               = "roles/iam.serviceAccountUser"
+  members = [
+    "serviceAccount:${google_service_account.cloud_deploy.email}",
+  ]
+}
+
+resource "google_clouddeploy_target" "prod" {
+  project     = var.project_id
+  provider    = google-beta
```

Comment:
This can be safely switched over to `google` provider. Looks like its already GA (https://github.com/hashicorp/terraform-provider-google/blob/2f09a571e647313d386a56b1aeaf87b4598fed49/CHANGELOG.md#4220-may-24-2022)